File: LeetCode_3542/Main.py
class Solution(object):
    def minOperations(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        st = []
        ans = 0

        for x in nums:
            if x == 0:
                st = []  # reset the stack
                continue

            # remove bigger elements
            while st and st[-1] > x:
                st.pop()

            # if new height (increasing point)
            if not st or st[-1] < x:
                ans += 1
                st.append(x)

        return ans

File: LeetCode_3542/Main.cpp
#include<bits/stdc++.h>
using namespace std;


class Solution {
public:
    int minOperations(vector<int>& nums) {
        vector<int> st;
        int ans = 0;

        for (int x : nums) {
            if (x == 0) {
                st.clear(); // reset
                continue;
            }

            while (!st.empty() && st.back() > x)
                st.pop_back();

            if (st.empty() || st.back() < x) {
                ans++;
                st.push_back(x);
            }
        }

        return ans;
    }
};
